src/main.rs

```diff
@@ -58,67 +58,49 @@ impl<H: Hasher + Default> BuildHasher for BuildDefaultHasher<H> {
 /// The tokens found will be passed to the given callback. The tokens will
 /// always be terminated by the delimiter at the end. Even if the last token
 /// is delimited by eof and not the delimiter, a delimiter will be supplied.
-fn split_read_zerocopy<R, F>(
-        delim: u8, inp: &mut R, mut handle_line: F)
-        -> Result<()>
-        where R: Read, F: FnMut(&[u8]) -> Result<()> {
-
+fn split_read_zerocopy<R, F>(delim: u8, inp: &mut R, mut handle_line: F) -> Result<()>
+where
+    R: Read,
+    F: FnMut(&[u8]) -> Result<()>,
+{
     // Initialize the buffer
     let mut buf = vec![0u8; pagesize() * 2]; // buf.capacity() == buf.len() == pagesize() * 2
 
-    // Amount of data actually in the buffer (manually managed so we
-    // can avoid reinitializing all the data when we resize)
-    let mut used: usize = 0;
-
     loop {
-        match inp.read(&mut buf[used..] /* unused space */) {
-            Err(ref e) if e.kind() == ErrorKind::Interrupted =>
-                // EINTR – thrown when a process receives a signal…
-                continue,
-            Err(e) =>
-                return Err(anyhow::Error::from(e)),
-            Ok(0) => {
+        // Amount of data actually in the buffer
+        let used = buf.len();
+        buf.resize(buf.capacity(), 0);
+        match inp.read(&mut buf[used..] /* unused space */)? {
+            0 => {
                 // EOF; we potentially need to process the last word here
```

Comment:
Does that comment still apply?

Reply:
Yes. If not `buf.is_empty()`, the last word is still in the buffer.